newton/_src/solvers/semi_implicit/kernels_body.py

```diff
@@ -34,36 +33,33 @@
 def joint_force(
     q: float,
     qd: float,
-    act: float,
+    joint_target_pos: float,
+    joint_target_vel: float,
     target_ke: float,
     target_kd: float,
     limit_lower: float,
     limit_upper: float,
     limit_ke: float,
     limit_kd: float,
-    mode: wp.int32,
 ) -> float:
     """Joint force evaluation for a single degree of freedom."""
 
     limit_f = 0.0
     damping_f = 0.0
     target_f = 0.0
 
-    if mode == JointMode.TARGET_POSITION:
-        target_f = target_ke * (act - q) - target_kd * qd
-    elif mode == JointMode.TARGET_VELOCITY:
-        target_f = target_ke * (act - qd)
+    target_f = target_ke * (joint_target_pos - q) + target_kd * (joint_target_vel - qd)
 
     # compute limit forces, damping only active when limit is violated
     if q < limit_lower:
         limit_f = limit_ke * (limit_lower - q)
         damping_f = -limit_kd * qd
-        if mode == JointMode.TARGET_VELOCITY:
-            target_f = 0.0  # override target force when limit is violated
+        if target_ke == 0.0:
+            target_f = 0.0
```

Comment:
This if condition right now doesn't do anything useful.
I think it makes sense to set `target_f` to zero regardless of `target_ke`, I think it helps with stability to disable the target force regardless when joint limits are violated.